### PR TITLE
Modify the library's suffix according the platform

### PR DESCRIPTION
**Pull request by zhaowanli2**
1. Modify the library's suffix according the platform; 
2. Create the menu when compile elastos; 
3. Update the android sample: HelloCarDemo.

### CODE REVIEW

File: Setup/SetEnv.sh [hunk 1]
```diff
@@ -1,3 +1,46 @@
...
+select opt in "${options[@]}"
+do
+    case $opt in
+        "linux")
+            echo "You chose [linux]"
+            SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+            export _ELASTOS64=YES
+            source $SCRIPT_DIR/SetEnv.sh x86_linux
+            export LD_LIBRARY_PATH=.
+            # source Ubuntu_SetEnv_RT.sh
```
Comment (joelliang) [inline Setup/SetEnv.sh]: Remove this comment

File: Setup/SetEnv.sh [hunk 2]
```diff
@@ -1,3 +1,46 @@
...
+#The arg is NULL, give user a choice.
+if [ "$1" == "" ]; then
+PS3='Please enter your choice: '
+options=("linux" "arm_android" "devtools_32" "devtools_64" "quit")
+select opt in "${options[@]}"
+do
+    case $opt in
+        "linux")
+            echo "You chose [linux]"
+            SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (joelliang) [inline Setup/SetEnv.sh]: You could move this line to the beginning of this file.

File: Setup/SetEnv.sh [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+            export _ELASTOS64=YES
+            source $SCRIPT_DIR/SetEnv.sh x86_linux
+            export LD_LIBRARY_PATH=.
+            # source Ubuntu_SetEnv_RT.sh
+            return;;
+        "arm_android")
+            echo "You chose [arm_android]"
+            SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (joelliang) [inline Setup/SetEnv.sh]: Remove this line as you will move the line 12 to the beginning of this file

File: Setup/SetEnv.sh [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+            source $SCRIPT_DIR/SetEnv.sh arm_android
+            return;;
+        "devtools_32")
+            echo "You chose [Devtools_32]"
+            SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (joelliang) [inline Setup/SetEnv.sh]: Remove it

File: Setup/SetEnv.sh [hunk 5]
```diff
@@ -1,3 +1,46 @@
...
+            source $SCRIPT_DIR/SetEnv.sh gcc_devtools
+            export LD_LIBRARY_PATH=.
+            return;;
+        "devtools_64")
+            echo "You chose [Devtools_64]"
+            SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
```
Comment (joelliang) [inline Setup/SetEnv.sh]: Remove it

File: Sources/ToolChains/Carl/ElCar/Lube/lubase/memstr.cpp [hunk 6]
```diff
@@ -80,7 +80,11 @@ int LubeContext::LibraryMember(MemberType member, char *pszBuffer)
         p--;
 
     if (Member_Name == member) {
-        while (*p && strcasecmp(".eco", p) != 0) *pszBuffer++= *p++;
+        char* suffix = ".eco";
+#ifdef _android
```
Comment (songsjun) [inline Sources/ToolChains/Carl/ElCar/Lube/lubase/memstr.cpp]: #ifdef _android
char* suffix =".so";
#else
char* suffix = ".eco";
#endif

**Pull request merged**